includes/abstracts/abstract-batch.php

```diff
@@ -271,8 +318,20 @@ public function run( $current_step ) {
 		$per_page = apply_filters( 'loco_batch_' . $this->slug . '_per_page', $per_page );
 
 		$total_steps = ceil( $this->total_num_results / $per_page );
+
 		if ( (int) $this->current_step === (int) $total_steps ) {
-			$this->update_status( 'finished' );
+
+			// Need to really check to make sure there were no results added while processing.
+			// In the case of destructive actions (i.e. deletion) there will be a gap equal to the per_page param.
+			// In all other cases, the difference in totals should equal total number of results.
+			// If neither of these are true, we need to run the last step over again.
+			$difference = $this->total_num_results - $this->difference_in_result_totals;
```

Comment:
Your comment mentions "If neither of these are true, we need to run the last step over again.". How do we ensure that we aren't processing records again by rerunning the same step in this case?